internal/artifacts/flowerofparadiselost/flowerofparadiselost.go

```diff
@@ -0,0 +1,104 @@
+package flowerofparadiselost
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/pkg/core"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+	"github.com/genshinsim/gcsim/pkg/core/keys"
+	"github.com/genshinsim/gcsim/pkg/core/player/artifact"
+	"github.com/genshinsim/gcsim/pkg/core/player/character"
+	"github.com/genshinsim/gcsim/pkg/modifier"
+)
+
+const (
+	icdKey = "flower-4pc-icd"
+	icd    = 60 // 1s
+
+	stackKey = "flower-4pc-stack-%v"
+)
+
+func init() {
+	core.RegisterSetFunc(keys.FlowerOfParadiseLost, NewSet)
+}
+
+type Set struct {
+	Index int
+}
+
+func (s *Set) SetIndex(idx int) { s.Index = idx }
+func (s *Set) Init() error      { return nil }
+
+// 2pc - Increases Elemental Mastery by 80.
+// 4pc - The equipping character's Bloom, Hyperbloom, and Burgeon reaction DMG are increased by 40%. Additionally, after the equipping
+//       character triggers Bloom, Hyperbloom, or Burgeon, they will gain another 25% bonus to the effect mentioned prior. Each stack
+//       of this lasts 10s. Max 4 stacks simultaneously. This effect can only be triggered once per second. The character who equips
+//       this can still trigger its effects when not on the field.
+func NewSet(c *core.Core, char *character.CharWrapper, count int, param map[string]int) (artifact.Set, error) {
+	s := Set{}
+
+	if count >= 2 {
+		m := make([]float64, attributes.EndStatType)
+		m[attributes.EM] = 80
+		char.AddStatMod(character.StatMod{
+			Base:         modifier.NewBase("flower-2pc", -1),
+			AffectedStat: attributes.EM,
+			Amount: func() ([]float64, bool) {
+				return m, true
+			},
+		})
+	}
+
+	if count >= 4 {
+		char.AddReactBonusMod(character.ReactBonusMod{
+			Base: modifier.NewBase("flower-4pc", -1),
+			Amount: func(ai combat.AttackInfo) (float64, bool) {
+				switch ai.AttackTag {
+				case combat.AttackTagBloom:
+				case combat.AttackTagHyperbloom:
+				case combat.AttackTagBurgeon:
+				default:
+					return 0, false
+				}
+
+				stacks := 0
+				for i := 0; i < 4; i++ {
+					key := fmt.Sprintf(stackKey, i+1)
+					if char.StatusIsActive(key) {
+						stacks++
+					}
+				}
+				return 0.4 * (1 + float64(stacks)*0.25), false
+			},
+		})
+
+		f := func(args ...interface{}) bool {
+			atk := args[1].(*combat.AttackEvent)
+			if atk.Info.ActorIndex != char.Index {
+				return false
+			}
+			if char.StatusIsActive(icdKey) {
+				return false
+			}
+			char.AddStatus(icdKey, icd, true)
+
+			// TODO: refresh stack?
```

Comment:
stacks don't have separate timers, so you need to rewrite this and the ReactBonusMod